File: app/src/main/java/com/example/mvx/mvx/android/guide/learn/project/annotation/InjectData.java
package com.example.mvx.mvx.android.guide.learn.project.annotation;

import android.app.Activity;
import android.text.TextUtils;

import java.lang.reflect.Field;

public class InjectData {

    public static void inject(Activity activity) throws IllegalAccessException {
        Class<? extends Activity> cls = activity.getClass();
        // 获取Activity中所有的成员属性
        Field[] declaredFields = cls.getDeclaredFields();
        for (Field filed: declaredFields) {
            // 判断是否该成员属性是否有 @InjectIntent 注解
            if (filed.isAnnotationPresent(InjectIntent.class)) {
                InjectIntent intent = filed.getAnnotation(InjectIntent.class);
                // 获取注解中传入的value
                String dataName = intent.value();
                System.out.println("dataName: " + dataName);
                String typeName = filed.getType().getSimpleName();
                System.out.println("typeName: " + typeName);
                if (!TextUtils.isEmpty(dataName)) {
                    // 设置权限，防止private不能修改
                    filed.setAccessible(true);
                    switch (typeName) {

                        case "int":
                            System.out.println("int:" + activity.getIntent().getIntExtra(dataName, 0));
                            filed.set(activity, activity.getIntent().getIntExtra(dataName, 0));
                            break;

                        case "String":
                            System.out.println("String: " + activity.getIntent().getStringExtra(dataName));
                            filed.set(activity, activity.getIntent().getStringExtra(dataName));
                            break;

                        case "boolean":
                            System.out.println("boolean: " + activity.getIntent().getBooleanExtra(dataName, false));
                            filed.set(activity, activity.getIntent().getBooleanExtra(dataName, false));
                            break;

                        default:
                            break;
                    }
                }
            }
        }
    }
}

File: app/src/main/java/com/example/mvx/mvx/android/guide/learn/project/annotation/InjectData.kt
package com.example.mvx.mvx.android.guide.learn.project.annotation

import android.app.Activity
import android.text.TextUtils

object InjectData {
    @JvmStatic
    @Throws(IllegalAccessException::class)
    fun inject(activity: Activity) {
        val cls: Class<out Activity> = activity.javaClass
        // 获取Activity中所有的成员属性
        val declaredFields = cls.declaredFields
        for (filed in declaredFields) {
            // 判断是否该成员属性是否有 @InjectIntent 注解
            if (filed.isAnnotationPresent(InjectIntent::class.java)) {
                val intent = filed.getAnnotation(InjectIntent::class.java)
                // 获取注解中传入的value
                val dataName: String = intent.value
                println("dataName: $dataName")
                val typeName = filed.type.simpleName
                println("typeName: $typeName")
                if (!TextUtils.isEmpty(dataName)) {
                    // 设置权限，防止private不能修改
                    filed.isAccessible = true
                    when (typeName) {
                        "int" -> {
                            println("int:" + activity.intent.getIntExtra(dataName, 0))
                            filed[activity] = activity.intent.getIntExtra(dataName, 0)
                        }

                        "String" -> {
                            println("String: " + activity.intent.getStringExtra(dataName))
                            filed[activity] = activity.intent.getStringExtra(dataName)
                        }

                        "boolean" -> {
                            println("boolean: " + activity.intent.getBooleanExtra(dataName, false))
                            filed[activity] = activity.intent.getBooleanExtra(dataName, false)
                        }

                        else -> {}
                    }
                }
            }
        }
    }
}